scipy/sparse/sparsetools/csr.h

```diff
@@ -704,6 +704,11 @@ void csr_binop_csr_general(const I n_row, const I n_col,
 
     I nnz = 0;
     Cp[0] = 0;
+    bool addsub = false;
+
+    //checks the type of binary operation to be performed.
+    if((int)op(8, 4) == 12 || (int)op(8, 4) == 4)
```

Comment:
Easy method to check if the binary operation is plus or minus.

Reply:
I'd much prefer defining a separate function for addition/subtraction, rather than special casing logic here.

Reply:
If want the explicit check to work for multiplication and division as well we have to split this into three functions. Two functions for arithmetic binary operations and another one for relational operations.

Reply:
Even I was doubtful about this but I wanted to keep the code small that's why I choose this special casing logic here. Anyways I have made three separate functions now . One will handle relational operations and other two will handle the arithmetic operations  .